src/grafana_dashboards/synapse.json

```diff
@@ -521,8 +525,11 @@
           {
             "datasource": "${prometheusds}",
             "editorMode": "code",
-            "expr": "rate(process_cpu_user_seconds_total{juju_application=\"synapse\"}[$__rate_interval]) + rate(process_cpu_system_seconds_total{juju_application=\"synapse\"}[$__rate_interval])",
-            "legendFormat": "__auto",
+            "expr": "avg(rate(process_cpu_seconds_total{instance=\"$instance\",juju_application=~\"$juju_application\",juju_model=~\"$juju_model\",juju_model_uuid=~\"$juju_model_uuid\",juju_unit=~\"$juju_unit\"}[5m]) * 1000)",
```

Comment:
Sure, is in seconds because process_cpu_seconds_total is a counter metric for the total user and system CPU time spent in seconds.

This could be read as the average number of 'CPU seconds' consumed per second over the 5-minute interval.

Similar to "rate(http_requests_total{job="api-server"}[5m])" but in this case you couldn't use "seconds" unit in the panel because is the number of 'requests' per second over the 5-minute interval.

Reply:
Ok, I think what I find confusing is that the title and description give a total when its a rate. I would normally expect a title like "Avg CPU time / sec" and a description that indicates the rate. For example here

![image](https://github.com/user-attachments/assets/13c8245d-fb41-4a57-ae73-aa71f186560b)


 Can you also explain the * 1000 multiplier? Thank you.

Reply:
Good catch, there is no need to multiply since Grafana unit is already set as seconds!

Reply:
Changed description and remove the 1000.

Reply:
I think what we are getting is CPU utilisation (because if you have 1 cpu second per second, that means you are using 100% of the cpu), see also https://github.com/canonical/synapse-operator/blob/a823bbfaa95afddcf22d7cb51b257860340df6ce/src/prometheus_alert_rules/synapse.rule#L32-L38 . It might be worth multiplying by 100 and displaying percentages and calling the panel "CPU Utilisation" (so something like what is mentioned in https://zhimin-wen.medium.com/custom-prometheus-metrics-for-apps-running-in-kubernetes-498d69ada7aa (section 4).

Reply:
This does not consider the amount of cores and that's why I'm doing this change.

Reply:
Since I have approvals and I'll close this PR and create a new one for removing the alert (and also review the others).

Reply:
Story created to address that: ISD-2334

Reply:
Hi @amandahla. I saw that https://github.com/element-hq/synapse/blob/d844afdc2970f772589b09e615c757d25645fda9/synapse/metrics/__init__.py#L367 are gauges, can the issue you have seen before (and the reason for this PR) be related to this (`rate` function had been applied to them)? 

`process_cpu_user_seconds_total` and `process_cpu_system_seconds_total` are gauges defined in the synapse application itself, whereas `process_cpu_seconds_total` comes from the prometheus client library: https://github.com/prometheus/client_python/blob/master/prometheus_client/process_collector.py#L75 (and correctly uses the recommended `_total` prefix). This may also explain the problem you reported in https://github.com/element-hq/synapse/issues/17605.